fix(ci): semgrep, cov-comment, buf-break-no-fail

MaxMustermann2 opened the pull request:
In the last push to the develop branch, semgrep [complained](https://github.com/ExocoreNetwork/exocore/actions/runs/8243299667/job/22543646196) about minor things: docker compose security attributes (fixed), a mutex not being unlocked (which has now been added to the list ignored files) and the conversion of uint64 to uint32 (fix taken from 26d7b72efd7b7d5c028bc015079d7d016590bddd).

An edit has been made to the `test.yml` workflow by splitting the coverage and comment workflows into two parts, thereby fixing a [potential security hole](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/).

Lastly, this PR silences the failure of buf's breaking check instead posting it as a comment to the PR. Since we are in the phase of rapid prototyping and testing, these failures add unneeded noise to all of the pull requests. However, the information provided by the workflow is still relevant, and hence, its status is posted as a comment to the pull request instead.

MaxMustermann2 committed:
chore(lint): security options in docker compose

MaxMustermann2 committed:
chore(lint): unlock mutex in testutil

MaxMustermann2 committed:
chore(lint): limit value of uint using strconv

MaxMustermann2 committed:
fix(lint): check out old code correctly for diff

In the event of a pull_request_target action, the checkout action by
default does not check out the PR but the base repository. This change
explicitly sets it up to check out the PR's head branch.

MaxMustermann2 committed:
Revert "fix(lint): check out old code correctly"

This reverts commit 4f24ba223548593149602c2bb8842c38fab2dd13.

MaxMustermann2 committed:
chore(lint): split test coverage commenter

See comment on test.yml for explanation.

MaxMustermann2 committed:
chore(ci): trigger go based CIs

delete superfluous comment

MaxMustermann2 committed:
fix(lint): add empty line at end of yaml file

MaxMustermann2 committed:
chore(lint): fix the test-comment workflow

- shell check the script
- add top level permissions

TimmyExogenous commented on testutil/network/network.go:
```diff
@@ -218,6 +218,7 @@ func New(l Logger, baseDir string, cfg Config) (*Network, error) {
 	// only one caller/test can create and use a network at a time
 	l.Log("acquiring test network lock")
 	lock.Lock()
+	defer lock.Unlock()
```
This might deviate from the original purpose of this function. What it aims to do is to unlock through `Cleanup` by the caller. It's a global lock. 
I also encountered this error when I ran `semgrep` locally. I am considering whether we should delete the  `testutil/network` directory since we don't use it now. or add the `lock.Unlock` in this function to eliminate the error, just like you did.

MaxMustermann2 replied:
I have added the `testutil` folder to `.semgrepignore`.

MaxMustermann2 committed:
Revert "chore(lint): unlock mutex in testutil"

This reverts commit a6cdb02745b6afe7d7c60277895921e5654d9631. It is said
that the unlocking of the mutex might deviate from the original purpose
of the function. Instead, we are electing to ignore the testutil folder,
which will be sent in the next commit.

MaxMustermann2 committed:
chore(ci): ignore testutil for semgrep

MaxMustermann2 committed:
chore(ci): do not fail if breaking proto changes

MaxMustermann2 committed:
chore(ci): trigger buf-breaking-action check

This commit will be reverted before merge of the PR.

MaxMustermann2 committed:
chore(ci): add comment if buf reports change

MaxMustermann2 committed:
fix(ci): only overwrite prev coverage comment

MaxMustermann2 committed:
fix(ci)(proto): +comment read/write permission

MaxMustermann2 committed:
fix(ci): add comment rw permission to job

MaxMustermann2 committed:
fix(ci): update job name in test-comment

MaxMustermann2 committed:
fix(ci): give `proto.yml` all rw permissions

MaxMustermann2 committed:
fix(ci): split buf-breaking and its comment

As before, the `pull_request` event does not have enough permissions to
make a comment and the `pull_request_target` cannot safely check out the
PR. Hence, we split the commenting process into 2 parts:
 - Run the buf breaking action and do not fail the workflow even if it
   errors
 - In a separate workflow, consume the artifact produced by the previous
   workflow and comment (or append to an existing comment) its contents.
Note that the *-comment.yml workflows are being commented blindly at the
moment because it is not possible to run them until they exist on the
branch.

MaxMustermann2 committed:
fix(ci): if condition shouldn't be always true

MaxMustermann2 commented:
Currently, there is a change in `proto/exocore/dogfood/v1/dogfood.proto` reported by the break-check action as a breaking change. However, the run has passed since we set `continue-on-error` to true. The result of the workflow has been uploaded as an artifact. The artifact contains the PR number as well.

There is a downside to this approach that the result is not immediately visible on the PR's front page (although it is visible within the files changed tab). To work around this problem, I set up another workflow to comment its result. When  `proto-comment.yml` is merged to `develop` it will ideally start running in response to (and after) `proto.yml` automatically.

![image](https://github.com/ExocoreNetwork/exocore/assets/82761650/cebeb7bb-e4da-4d86-80e9-c3e3618111af)

MaxMustermann2 committed:
Revert "chore(ci): trigger go based CIs"

This reverts commit 62b3744cc65d994d28c0c9ef69c06e0d4239bb8e.

MaxMustermann2 committed:
Revert "chore(ci): trigger buf-breaking-action...

"check." This reverts commit 88566fdfb4ed74935374c4062b950f35beaf8e4b.

TimmyExogenous reviewed (approved):
Great work.

adu-web3 reviewed (approved):
LGTM 👍
